feat(babel-plugin-component): remove import dedupe

wjhsf opened the pull request:
## Details

`@lwc/babel-plugin-component` does import deduping, but this is unnecessary as it is also done by rollup. This is a potentially breaking change if code relies on the order of imports. However, it only impacts direct consumers of `@lwc/babel-plugin-component` and `@lwc/compiler`. It does not impact consumers of `@lwc/rollup-plugin` or users of a bundler (like rollup) that does deduping.

Closes #5023.

## Does this pull request introduce a breaking change?

<!--
    Any change that can cause downstream consumers to fail qualifies as a breaking change.

    Examples:
        - Removing the code for a deprecated API.
        - Adding a new restriction to the compiler which might result in a compilation failure for existing code.
        - Changing the return type of a function in a non-backward compatible fashion.

    Remove the incorrect item for the list.
-->

- 😮‍💨 No, it does not introduce a breaking change.
- 💔 Yes, it does introduce a breaking change.

<!-- If yes, please describe the impact and migration path for existing applications. -->

## Does this pull request introduce an observable change?

<!--
    Observable changes are internal changes that can be observed by downstream consumers.
    Such changes don't qualify as breaking changes because they don't impact any publicly defined
    APIs.

    Examples:
        - Fixing a bug.
        - Changing the invocation timing of a callback, for a callback that has no invocation timing
          guarantee.

    Remove the incorrect item from the list.
-->

- 🤞 No, it does not introduce an observable change.
- 🔬 Yes, it does include an observable change.

<!-- If yes, please describe the anticipated observable changes. -->

## GUS work item

<!-- Work ID in text, if applicable. -->

jye-sf reviewed (changes requested):
The description mentions that direct consumers of @lwc/compiler will also be impacted by this change. How will that be affected, and can we get tests for that?

wjhsf commented on packages/@lwc/rollup-plugin/src/__tests__/importDedupe/importDedupe.spec.ts:
```diff
@@ -0,0 +1,50 @@
+    expect(imports).toHaveLength(1);
+    const specifiers = imports[0][1];
+    expect(specifiers).toContain('renderer');
+    expect(specifiers).not.toContain('renderer as');
```
The relevant output is `import { LightningElement, renderer } from 'lwc'`. The multiple imports in `basic.js` should be merged and they should not use `as`. We don't check for that exact string because there are other imports added as part of compilation (`registerComponent`) and we don't care about the ordering.

wjhsf replied:
added a code comment to clarify

wjhsf commented:
I took a deeper look at the compiler, and here are the relevant transformations that it does:
1. If it's transforming a component file (`export default class extends LightningElement {}`), then it inserts a template import (`import _html from "./test.html"`) at the top of the file.
2. If a decorator is used, it adds `import { registerDecorators as _registerDecoratores } from 'lwc'` above `import _tmpl`.
3. If a component feature flag is used, it adds the feature flag import (`import featureFlag from '@salesforce/featureFlag/FLAG_NAME`) above `import _tmpl`. 
    a. This is a new feature and the imported module is most likely unique, so I don't think we need to consider it for dedupe breakage. 
4. Previously, all `import` statements were merged into the first occurrence of each imported module. Now they are left as written. 

```js
// actual.js
import { bar } from "foo";
globalThis.modified = true;
import { qux } from "foo";

// expected.js before
import { bar, qux } from "foo";
globalThis.modified = true;

// expected.js after
import { bar } from "foo";
globalThis.modified = true;
import { qux } from "foo";
```

Note that the order of imports only matters if the exports rely on a global state _and_ it's loaded in a non-standard way. Both ESM `import` and CJS `require` leverage caching, so that the imported module is only evaluated once.

🤔 All synthetic shadow tests are currently failing. That's sus...
